cmd/simpleargs/main.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+
+	"github.com/google/go-github/github"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var regex string
+var eventfile string
+var writepath string
+
+type roundTripper struct {
+	accessToken string
+}
+
+func (rt roundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	r.Header.Set("Authorization", fmt.Sprintf("token %s", rt.accessToken))
+	return http.DefaultTransport.RoundTrip(r)
+}
+func writeArgs(groups []string) {
+	for i, group := range groups[1:] {
+		data := []byte(group)
+		filename := fmt.Sprintf("ARG_%d", i)
+		log.Printf("In write args")
+		err := ioutil.WriteFile(filepath.Join(writepath, filename), data, 0644)
+		log.Printf(filepath.Join(writepath, filename))
+		if err != nil {
+			log.Fatalln(err)
+		}
+	}
+}
+
+func writeArg(groups string, filetype string) {
+	data := []byte(groups)
+	filename := fmt.Sprintf("ARG_%s", filetype)
+	log.Printf("In write arg")
+	err := ioutil.WriteFile(filepath.Join(writepath, filename), data, 0644)
+	log.Printf(filepath.Join(writepath, filename))
+	if err != nil {
+		log.Fatalln(err)
+	}
+}
+
+func main() {
+	app := kingpin.New(filepath.Base(os.Args[0]), "simpleargs github comment extract")
+	app.Flag("eventfile", "path to event.json").Default("/github/workflow/event.json").StringVar(&eventfile)
+	app.Flag("writepath", "path to write args to").Default("/github/home").StringVar(&writepath)
+	app.Arg("regex", "Regex pattern to match").Required().StringVar(&regex)
+	kingpin.MustParse(app.Parse(os.Args[1:]))
+	log.Printf("After flag declaration")
+	var err error
+
+	//Github client for posting comments
+	token := os.Getenv("GITHUB_TOKEN")
+	http.DefaultClient.Transport = roundTripper{token}
+	client := github.NewClient(http.DefaultClient)
+
+	//Decoding and saving service account authentication file
```

Comment:
In Secrets, we store the base64 encoded auth file. I decoded and saved it here instead of having a separate action for doing so

Reply:
I also think that we don't need this?  
I think the prombench tool can read env variables so it will be something like: `/prombench .... --auth=$AUTH_FILE` if that is not the case we can extend it to support this format instead of just a file.